akka-http-core/src/main/scala/akka/http/scaladsl/Http.scala

```diff
@@ -386,6 +386,31 @@ class HttpExt private[http] (private val config: Config)(implicit val system: Ex
   private[http] val poolMaster: PoolMaster = PoolMaster()
   private[this] val systemMaterializer = SystemMaterializer(system).materializer
 
+  /**
+   * Creates a builder which will create a single connection to a host every time the built flow is materialized. There
+   * is no pooling and you are yourself responsible for lifecycle management of the connection. For a more convenient
+   * Request level API see [[singleRequest()]]
+   *
+   * The responses are not guaranteed to arrive in the same order as the requests go out (In the case of a HTTP/2 server)
+   * so therefore requests needs to have a [[akka.http.scaladsl.model.http2.RequestResponseAssociation]]
+   * which Akka HTTP will carry over to the corresponding response for a request.
+   *
+   *
+   * @return A builder to configure more specific setup for the connection and then build a `Flow[Request, Response, Future[OutgoingConnection]]`.
+   */
+  /*
+   * FIXME: I think we'd want this in the HTTP extension for easy access but it cannot refer to classes in the http2-support module
+   * FIXME: for having the same entry point for HTTP/2 and HTTP/1.1 there is a mismatch with port always reasonably defaulting to 443
+   *        for HTTP/2 but not really for HTTP1/1
+   * FIXME also for same API and automatic protocol negotiation, how do we deal with explicitly saying TLS/No TLS for HTTP/1 but defaulting to TLS for HTTP/2?
+   */
+  def connectionTo(host: String): OutgoingConnectionBuilder =
```

Comment:
These `FIXME`'s and the actual location of `connectionTo` (`Http` instead of `Http2`) confused me a bit regarding how this will look in the end.

Reply:
I was thinking a user would use `val flow: Flow[] = Http2().someMethods()...` to connect to a server using HTTP/2 (over TLS or WithPriorKnowledge).

Alternatively, they would use `val flow: Flow[] = Http().someMethods()...` to connect to a server using HTTP/1.x (over TLS or cleartext).

Using `Http2()` or `Http()` would be enough for the user to be aware that the obtained `Flow` is ordered or not.

Reply:
See the comment above regarding private Http2 vs. public Http.